web/src/app/shared/store/classification-store/classification.state.ts

```diff
@@ -118,33 +113,59 @@ export class ClassificationState implements NgxsAfterBootstrap {
+    // returns true in case the classification was edited or copied
+    if (state.classifications.some(classification => classification.classificationId === action.classificationId)) {
+      return this.classificationsService.getClassification(action.classificationId).pipe(
+        take(1),
+        tap(selectedClassification => {
+          ctx.patchState({ selectedClassification });
+        })
+      );
+    }
+
+    // the classification is restored to a new classification
+    const category = state.classificationTypes[state.selectedClassificationType][0];
+    const { type, created, modified, domain, parentId, parentKey } = state.selectedClassification;
+    ctx.patchState({ selectedClassification: { type, created, category, modified, domain, parentId, parentKey } });
+
+    return of(null);
   }
 
-  @Action(SetActiveAction)
-  setActiveAction(ctx: StateContext<ClassificationStateModel>, action: SetActiveAction): Observable<null> {
-    if (action.action === ACTION.CREATE) {
-      // Initialization of a new classification
-      const state: ClassificationStateModel = ctx.getState();
-      const date = TaskanaDate.getDate();
-      const initialClassification: Classification = {
-        type: state.selectedClassificationType,
-        category: state.classificationTypes[state.selectedClassificationType][0],
-        created: date,
-        modified: date,
-        domain: this.domainService.getSelectedDomainValue(),
-      };
-      if (state.selectedClassification) {
-        initialClassification.parentId = state.selectedClassification.classificationId;
-        initialClassification.parentKey = state.selectedClassification.key;
-      }
-      ctx.patchState({ selectedClassification: initialClassification, action: action.action });
-    } else {
-      ctx.patchState({ action: action.action });
+  @Action(CreateClassification)
+  newCreateClassification(ctx: StateContext<ClassificationStateModel>): Observable<null> {
+    // Initialization of a new classification
+    const state: ClassificationStateModel = ctx.getState();
+    const date = TaskanaDate.getDate();
+    const initialClassification: Classification = {
+      type: state.selectedClassificationType,
+      category: state.classificationTypes[state.selectedClassificationType][0],
+      created: date,
+      modified: date,
+      domain: this.domainService.getSelectedDomainValue(),
+    };
+    if (state.selectedClassification) {
+      initialClassification.parentId = state.selectedClassification.classificationId;
+      initialClassification.parentKey = state.selectedClassification.key;
     }
+    ctx.patchState({
+      selectedClassification: initialClassification,
+      badgeMessage: 'Creating new classification'
+    });
+    return of(null);
+  }
+
+  @Action(CopyClassification)
+  newCopyClassification(ctx: StateContext<ClassificationStateModel>): Observable<null> {
+    const copy = { ...ctx.getState().selectedClassification };
+    copy.key = null;
```

Comment:
why key = null, but classificationId = undefined?
i think classificationId should be null too, as we deliberatly set it to nothing

Reply:
other attributes are copied from the selected classification but id and key are not, so that user can write new key value, while id will be generated by backend when submitted. This behavior is similar to workbasket.

Reply:
yes, but shouldn't both be set to null, as null indicates the explicit absence of value, while undefined should show, that the value has not been initiated

Reply:
Setting the id to null was also my first intuition but this leads to problems in the tree component since the classificationId is used as idField for the tree nodes.